sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c

```diff
@@ -352,13 +352,31 @@ static int _fsm_call_close(gnrc_tcp_tcb_t *tcb)
 /**
  * @brief FSM handling function for forcefull connection teardown sequence.
  *
- * @returns   -EOPNOTSUPP. Currently not implemented.
+ * @param[in,out] tcb   TCB holding the connection information.
+ *
+ * @returns   Zero on success.
  */
-static int _fsm_call_abort(void)
+static int _fsm_call_abort(gnrc_tcp_tcb_t *tcb)
 {
+    gnrc_pktsnip_t *out_pkt = NULL;     /* Outgoing packet */
+    uint16_t seq_con = 0;               /* Sequence number consumption of outgoing packet */
+
     DEBUG("gnrc_tcp_fsm.c : _fsm_call_abort()\n");
-    DEBUG("gnrc_tcp_fsm.c : _fsm_call_abort() : ABORT not implemented\n");
-    return -EOPNOTSUPP;
+
+    /* A reset must be sent in case the TCB state is in one of those cases */
+    if (tcb->state == FSM_STATE_SYN_RCVD || tcb->state == FSM_STATE_ESTABLISHED ||
```

Comment:
yeah right, not sure if that matters here - and who knows what modern compiler optimize away anyway 😉 My point rather was, that it might be more readable and extensible (if required in the future) using switch-case over if-else.

But as said, not in this PR - as it is not related ...